pkgs/tools/system/btop/default.nix

```diff
@@ -37,6 +39,11 @@ stdenv.mkDerivation rec {
     ${removeReferencesTo}/bin/remove-references-to -t ${stdenv.cc.cc} $(readlink -f $out/bin/btop)
   '';
 
+  postPhases = lib.optionals rocmSupport [ "postPatchelf" ];
+  postPatchelf = lib.optionals rocmSupport ''
+    patchelf --set-rpath "${rocmPackages.rocm-smi}/lib:$(patchelf --print-rpath "$out/bin/btop")" $out/bin/btop
```

Comment:
AFAIK `patchelf` in nixpkgs does come with the `--add-rpath` option so you can avoid all the bash expansions